lib/notifications/navigate.dart

```diff
@@ -0,0 +1,317 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:collection/collection.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import '../api/model/model.dart';
+import '../generated/l10n/zulip_localizations.dart';
+import '../host/notifications.dart';
+import '../log.dart';
+import '../model/binding.dart';
+import '../model/narrow.dart';
+import '../widgets/app.dart';
+import '../widgets/dialog.dart';
+import '../widgets/message_list.dart';
+import '../widgets/page.dart';
+import '../widgets/store.dart';
+
+NotificationPigeonApi get _notifPigeonApi => ZulipBinding.instance.notificationPigeonApi;
+
+/// Service for handling notification navigation.
+class NotificationNavigationService {
```

Comment:
What's your thinking behind naming this class and this file with "navigate", rather than "open" as in the previous revision?

(Either name seems plausible to me; but I'm curious to hear what shifted your thinking to cause you to rename it.)

Reply:
(Re `NotificationNavigationData`, we'll probably want that to end up covering the more general "open" concept too: in particular we'll need the message ID from the notification, and that will naturally become another field on that class. That more general concept is reflected in the name of the existing `NotificationOpenPayload`, and in the dartdoc this version has which I supplied at https://github.com/zulip/zulip-flutter/pull/1379#discussion_r2094343971 above.)

Reply:
> For example, we might want to take you directly to the relevant message in history, rather than to the first unread or the newest message. (Maybe it was an @-mention in the middle of a long-running conversation that you weren't previously reading.)

(Just filed an issue for this: #1565.)